### PR TITLE
Fum 3224 dd k8s secrets

### PR DESCRIPTION
**Pull request by AlfGot**
### Changes
- replacing kubernetes secret creation by external secret
- adding new variable for DataDog secret name
The expected WF:
- The DD secret is already available in Secrets manager, created through https://github.com/tx-pts-dai/datadog-settings/blob/main/account-specific/terraform/keys.tf
- The secret name is passed to the module (default is the name we already use in the previous step).
- The external secret is created through the external secret operator and secret store.

### CODE REVIEW

File: modules/datadog/variables.tf [hunk 1]
```diff
@@ -46,3 +46,9 @@ variable "datadog_operator_helm_values" {
     },
   ]
 }
+
+variable "datadog_secret" {
+  description = "Name of the datadog secret in Secrets Manager"
+  type        = string
+  default     = "dai-datadog/tamedia/keys"
```
[the remark this answers is not in the document]
Comment (AlfGot) [reply modules/datadog/variables.tf]: Good point, thank you! 
Removed, but I still need to use it in the DD example in order to get it working.

**Review by swibrow — CHANGES_REQUESTED** [review]
You can remove the datadog provider now since its actually not used.

File: examples/datadog/main.tf [hunk 2]
```diff
@@ -125,5 +125,7 @@ module "datadog" {
 
   cluster_name = module.k8s_platform.eks.cluster_name
 
+  datadog_secret = "dai-datadog/tamedia/keys"
```
Comment (rbapst-tamedia) [inline examples/datadog/main.tf:128]:
```suggestion
  datadog_secret = "dai/datadog/tamedia/keys"
```
from https://tamedia.atlassian.net/wiki/spaces/PTS/pages/44304906/Secrets+Naming+convention+and+tags we should use "/"

File: modules/datadog/main.tf [hunk 3]
```diff
@@ -1,24 +1,4 @@
...
-resource "datadog_api_key" "datadog_agent" {
-  name = coalesce(var.datadog.agent_api_key_name, var.cluster_name)
-}
-
-resource "datadog_application_key" "datadog_agent" {
-  name = coalesce(var.datadog.agent_app_key_name, var.cluster_name)
-}
 
 locals {
   datadog_site = "datadoghq.eu"
```
Comment (rbapst-tamedia) [inline modules/datadog/main.tf:4]: here too, if it's a module we should have this as an input.

**Review by swibrow — COMMENTED** [review]
Also the kubernetes provider can be removed? Its not used in the root either.

File: examples/datadog/main.tf (hunk 2 above)
Comment (AlfGot) [reply examples/datadog/main.tf:128]: I've created a second secret the old one is marked as obsolete and will be deleted once conformed the new one works.

File: modules/datadog/main.tf (hunk 3 above)
Comment (AlfGot) [reply modules/datadog/main.tf:4]: This part has been removed, since we no longer create the keys.

Comment (AlfGot) [reply modules/datadog/main.tf:4]: This code has been removed since no longer used.

File: modules/datadog/versions.tf [hunk 4]
```diff
@@ -5,14 +5,6 @@ terraform {
...
-      source  = "hashicorp/kubernetes"
-      version = ">= 2.0.0"
-    }
-    datadog = {
-      source  = "DataDog/datadog"
-      version = "~> 3.39"
-    }
     helm = {
       source  = "hashicorp/helm"
       version = "~> 2.6"
```
Comment (swibrow) [inline modules/datadog/versions.tf:10]: >= its common practice to use this logic for modules.

**Review by swibrow — APPROVED** [review]
Looking good

**Pull request merged**